00_notebooks/R-client.ipynb

```diff
@@ -0,0 +1,434 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# R client for lakeFS\n",
+    "\n",
+    "This shows how to generate an R client for lakeFS from the published lakeFS API `swagger.yml`. \n",
+    "\n",
+    "**As of July 2023 the R client is not shipped nor supported by lakeFS.** See [#6177](https://github.com/treeverse/lakeFS/issues/6177) for more details. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate the R client using openapi"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Run this from the CLI in the root of the lakeFS source repository"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```bash\n",
+    "docker run --rm -v $PWD:/mnt openapitools/openapi-generator-cli:latest generate \\\n",
+    "\t\t\t\t --input-spec /mnt/api/swagger.yml \\\n",
+    "\t\t\t\t --generator-name r \\\n",
+    "\t\t\t\t --output /mnt/clients/r \\\n",
+    "\t\t\t\t --package-name lakefsclient \\\n",
+    "\t\t\t\t --http-user-agent \"lakefs-r-sdk/rmoff\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Build the R client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Run this from the CLI in the root of the lakeFS source repository"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```bash\n",
+    "docker run --tty --rm -v \"$PWD/clients/r\":/home/docker \\\n",
+    "           -w /home/docker \\\n",
+    "           r-base R CMD build .\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Make the R client available to the notebook"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Run this from the CLI in the root of the lakeFS source repository"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```bash\n",
+    "cp clients/r/lakefsclient_1.0.0.tar.gz ../lakeFS-samples/data\n",
```

Comment:
Can we generate something that's _not_ "1.0.0"?  If we ever decide to publish R clients this will matter.  Even a tag such as "local" would work here.

Reply:
👍 it doesn't accept alpha chars so gone for `0.0.0.1`